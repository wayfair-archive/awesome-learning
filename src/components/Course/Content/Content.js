import React from 'react';
import './Content.scss';

const Content = ({ body, title }) => (
  <div className="Content">
    <h1 className="Content-title">{title}</h1>
    <div className="Content-body" dangerouslySetInnerHTML={{ __html: body }} />
  </div>
);

export default Content;
