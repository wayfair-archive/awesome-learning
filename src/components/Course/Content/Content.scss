@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.Content {
  max-width: $layout-course-single-width;
  padding: 0 15px;
  margin: 0 auto;

  &-title {
    font-size: $font--1500;
    max-width: $layout-course-width;
    margin: auto;
    font-weight: 600;
    text-align: center;
    @include line-height(1.65);
    @include margin-top(1);
    @include margin-bottom(0)
  }

  &-body {
    & figure {
      @include margin-bottom(1);

      & blockquote {
        font-style: italic;
        text-align: center;
        margin-top: 0;
        @include padding(1, 0);

        & p {
          max-width: $layout-course-width;
          font-size: $font--1500;
          margin-top: 0;
          @include margin-bottom(1);
          @include line-height(1.5)
        }

      }

    }

    & a {
      text-decoration: underline
    }

    & * {
      max-width: $layout-course-width;
      margin-left: auto;
      margin-right: auto
    }

    & img {
      max-width: 100%;
    }

  }

}

@include breakpoint-md {
  .Content {
    padding: 0;

    &-title {
      font-size: $font--3500;
      @include line-height(2.25);
      @include margin-top(2.25);
      @include margin-bottom(1.5)
    }

    &-body {
      font-size: $font--1500;
      @include line-height(1.125);
      @include margin-bottom(1.125);

      & p {
        font-size: $font--1500;
        @include line-height(1.125);
        @include margin-bottom(1.125)
      }

    }

  }

}