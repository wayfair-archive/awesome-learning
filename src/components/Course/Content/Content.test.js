import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      title: 'test',
      body: '<p>test</p>'
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
