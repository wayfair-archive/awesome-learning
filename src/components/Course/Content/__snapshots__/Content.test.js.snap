// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="Content"
>
  <h1
    className="Content-title"
  >
    test
  </h1>
  <div
    className="Content-body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
